lib/report/account_events.rb

```diff
@@ -0,0 +1,65 @@
+module Report
+  class AccountEvents
+    SESSION_DURATION = 5.hours
+
+    def self.report(options)
+      new(options).report
+    end
+
+    attr_reader :start_date, :end_date
+
+    def initialize(start_date:, end_date:, user_id_pepper:)
+      @start_date = start_date
+      @end_date = end_date
+      @user_id_pepper = user_id_pepper
+    end
+
+    def report
+      previous_login_times = {}
+
+      all_login_events
+        .where("created_at >= ? AND created_at < ?", start_date, end_date)
```

Comment:
I saw a PR recently where we had a off by one condition with a similar query.
We'd used BETWEEN there, is that easier to grok?

Reply:
Ah yeah, https://github.com/alphagov/govuk-account-manager-prototype/pull/586/commits/994ded86ccb3e6a6a8c0d5ea69b83be649008d36 - I'll update this